doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+
+The last step of processing the inference data is drawing masks using the
+received contours. Honestly speaking, there is no need to get the most accurate
+contours in the world: we can always blur masks borders by gaussian filter to
+occupy a bit more area. Another point that should be mentioned here is getting
+three masks (for areas to be smoothed, for ones to be sharpened and for the
+background) which have no intersections with each other; this approach allows to
+apply the calculated masks to the corresponding images prepared beforehand and
+then just to summarize them to get the output image without any other actions.
+
+All the operations described above are supported fascinatingly by the G-API
+functionality, so this algorithm is appropriate to illustrate G-API usage
+convenience and efficiency in the context of solving a real CV/DL problem.
+
+# Preparing networks to be inferenced with G-API {#gapi_fb_prep_nets}
+
+Let's prepare our chosen networks for being processed inside the graph.
+Providing a new network support is quite similar to a new operation
+implementation.
+
+To declare a deep learning topologies we should use the specific macro
+`G_API_NET()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_decl
+
+Similarly to `G_API_OP()` macro for a new operation, it takes three arguments to
+register a new type.
+They are:
+1. Network interface name --- also serves as a name of a new type defined
+   with this macro;
+2. Network signature      --- an `std::function<>`-like signature which defines
+   inputs and outputs;
+3. Network's unique name  --- used to identify it within the system when the
+   type is stripped.
+
+# Graph compilation {#gapi_fb_comp}
+
+For further convenience we should use the version of a pipeline expression
+with a lambda-based constructor to keep all temporary objects in a dedicated
+scope (almost all the graph's body is skipped in this code snippet
+except for input and output nodes):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_4
+
+To apply a model inside the graph, we should call
+`cv::gapi::infer<>()` setting a network type as a template argument to
+specify the topology we want to infer:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_usg_fd
+
+If we need to convey several ROIs for inference, we can use
+the `cv::gapi::infer<>()` function's overload:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_usg_ld
+
+The second argument of it is the image to be processed, and the first is a
+`cv::GArray<>` of ROIs; the network will be automatically inferred for every single ROI.
```

Comment:
There's nothing about graph compilation. The code in generated doxygen looks very weird. I don't see pipeline code here.

Details about "lamda-based constructor" can be omitted.

Reply:
What is this section should be about? Can it be "Describing the pipeline"? Should I show the whole pipeline here?

Reply:
May be marks for snippets have been set incorrectly but I didn't see much code here while reviewing doxygen. The code I've seen was neither about construction nor about the compilation of the graph.

Putting a snippet on how a top-level pipeline is built would make a perfect sense here.